aten/src/ATen/native/DilatedMaxPool2d.cpp

```diff
@@ -181,59 +136,68 @@ Tensor& max_pool2d_with_indices_backward_out_cpu_template(
     nInputPlane,
     inputHeight, inputWidth,
     outputHeight_for_shape_check, outputWidth_for_shape_check,
-    input.suggest_memory_format());
+    memory_format);
 
-  max_pool2d_backward_kernel(kCPU, gradInput, gradOutput, indices);
-
-  return gradInput;
+  set_output(0, input.sizes(), {}, input.options().memory_format(memory_format), input.names());
 }
+} // namespace meta
 
-Tensor& max_pool2d_with_indices_backward_out_cpu(
-  const Tensor& gradOutput,
-  const Tensor& input,
-  IntArrayRef kernel_size,
-  IntArrayRef stride,
-  IntArrayRef padding,
-  IntArrayRef dilation,
-  bool ceil_mode,
-  const Tensor& indices,
-  Tensor& gradInput)
-{
-  max_pool2d_with_indices_backward_out_cpu_template(
-    gradInput,
-    gradOutput,
-    input,
-    indices,
-    kernel_size,
-    stride,
-    padding,
-    dilation,
-    ceil_mode);
-  return gradInput;
+namespace native {
+
+TORCH_IMPL_FUNC(max_pool2d_with_indices_out_cpu)
+(const Tensor& input,
+IntArrayRef kernel_size,
+IntArrayRef stride,
+IntArrayRef padding,
+IntArrayRef dilation,
+bool ceil_mode,
+const Tensor& output,
+const Tensor& indices) {
+  NoNamesGuard guard;
+
+  const int kH = safe_downcast<int, int64_t>(kernel_size[0]);
+  const int kW = kernel_size.size() == 1 ? kH : safe_downcast<int, int64_t>(kernel_size[1]);
+
+  const int dH = stride.empty() ? kH : safe_downcast<int, int64_t>(stride[0]);
+  const int dW = stride.empty() ? kW :
+                 stride.size() == 1 ? dH : safe_downcast<int, int64_t>(stride[1]);
+
+  const int padH = safe_downcast<int, int64_t>(padding[0]);
+  const int padW = padding.size() == 1 ? padH : safe_downcast<int, int64_t>(padding[1]);
+
+  const int dilationH = safe_downcast<int, int64_t>(dilation[0]);
+  const int dilationW = dilation.size() == 1 ? dilationH : safe_downcast<int, int64_t>(dilation[1]);
+
+  TORCH_CHECK((input.ndimension() == 3 || input.ndimension() == 4),
+    "non-empty 3D or 4D (batch mode) tensor expected for input");
+
+  TORCH_CHECK(input.dtype() == output.dtype(),
+    "expected dtype ", input.dtype(), " for `output` but got dtype ", output.dtype());
```

Comment:
You should be able to skip these tests now since they're done in the meta function

Reply:
ok, I will delete these  checks

Reply:
doesn't look deleted to me lol